unit_tests/la/schur_compl_test.cpp

```diff
@@ -135,7 +135,7 @@ TEST(schur, complement) {
 		const PetscScalar *vals;
 		for (unsigned int i=0; i<block_size; i++) {
 			MatGetRow(schurComplement->get_a_inv(), i + rank*block_size, &ncols, &cols, &vals);
-			EXPECT_FLOAT_EQ( (1.0 / (double)(rank + 2)), vals[i] );
+			EXPECT_FLOAT_EQ( (1.0 / (double)(rank + 2)), vals[0] );

```

Comment:
Why this change?

Reply:
Thanks to setting `MatSetOption(matrix_, MAT_IGNORE_ZERO_ENTRIES, PETSC_TRUE);`,
the matrix `get_a_inv()` is now sparser:
``` 
row 0: (0, 0.5) 
row 1: (1, 0.5)
```
therefore vector vals has NOT size==block_size, but only 1 (only the diagonal value is nonzero in the test)